src/types/serialized-type.ts

```diff
@@ -0,0 +1,64 @@
+import { BytesList } from '../serdes/binary-serializer'
+const { bytesToHex, slice } = require('../utils/bytes-utils')
+
+const Comparable = {
+  lt (other) {
+    return this.compareTo(other) < 0
+  },
+  eq (other) {
+    return this.compareTo(other) === 0
+  },
+  gt (other) {
+    return this.compareTo(other) > 0
+  },
+  gte (other) {
+    return this.compareTo(other) > -1
+  },
+  lte (other) {
+    return this.compareTo(other) < 1
+  }
+}
+
+const SerializedType = {
+  toBytesSink (sink) {
+    sink.put(this._bytes)
+  },
+  toHex () {
+    return bytesToHex(this.toBytes())
+  },
+  toBytes () {
+    if (this._bytes) {
+      return slice(this._bytes)
+    }
+    const bl = new BytesList()
+    this.toBytesSink(bl)
+    return bl.toBytes()
+  },
+  toJSON () {
+    return this.toHex()
+  },
+  toString () {
+    return this.toHex()
+  }
+}
+
+function ensureArrayLikeIs (Type, arrayLike) {
```

Comment:
This is pretty terrible naming - I have no idea what this function is doing. 

Are we trying to verify types? If so typescript should help us get rid of this.